integration-tests/docker/test_env/test_env.go

```diff
@@ -101,17 +102,18 @@ func (te *CLClusterTestEnv) WithPrivateChain(evmNetworks []blockchain.EVMNetwork
 	var chains []test_env.PrivateChain
 	for _, evmNetwork := range evmNetworks {
 		n := evmNetwork
-		pgc := test_env.NewPrivateGethChain(&n, []string{te.Network.Name})
-		if te.t != nil {
-			pgc.GetPrimaryNode().WithTestLogger(te.t)
-		}
-		chains = append(chains, pgc)
 		var privateChain test_env.PrivateChain
 		switch n.SimulationType {
 		case "besu":
 			privateChain = test_env.NewPrivateBesuChain(&n, []string{te.Network.Name})
+			if te.t != nil {
+				privateChain.GetPrimaryNode().WithTestLogger(te.t)
+			}
 		default:
 			privateChain = test_env.NewPrivateGethChain(&n, []string{te.Network.Name})
+			if te.t != nil {
+				privateChain.GetPrimaryNode().WithTestLogger(te.t)
+			}
 		}
 		chains = append(chains, privateChain)
```

Comment:
```suggestion
		if te.t != nil {
			privateChain.GetPrimaryNode().WithTestLogger(te.t)
		}
		chains = append(chains, privateChain)
```
instead of putting the same logic above twice in the switch statement.